src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -1334,6 +1334,20 @@ MavlinkReceiver::handle_message_odometry(mavlink_message_t *msg)
 		odometry.velocity_covariance[odometry.COVARIANCE_MATRIX_VY_VARIANCE] = lin_vel_cov_local(1, 1);
 		odometry.velocity_covariance[odometry.COVARIANCE_MATRIX_VZ_VARIANCE] = lin_vel_cov_local(2, 2);
 
+	} else if (odom.child_frame_id == MAV_FRAME_LOCAL_FRD) {
```

Comment:
This does not respect the convention specified for the ODOMETRY messages, which is inerited from the ROS Odometry msg (http://docs.ros.org/melodic/api/nav_msgs/html/msg/Odometry.html) - child frame ID should be a body frame.

Reply:
@TSC21 I completely agree, child frame ID should be a body frame and parent should be local or global. We don't use ROS, we're just trying to get odometry data into PX4 over mavlink. I'm very confused why the existing code expects odometry messages with position XYZ in local frame but velocity in body frame which would then need to be rotated back to local frame (lines 1302 - 1335).

I wanted to use a "body" child frame but there are no other options in the mavlink spec. I chose MAV_FRAME_LOCAL as an alternative so I didn't effect the existing code. Which frame should we use instead? or perhaps we should just have the velocity direct-copy without rotation as the default behavior in the else statement?